lib/add_contact.rb

```diff
@@ -84,12 +87,24 @@ def valid_email?
   end
 
   def click
-    Contacts.create(:first_name => find_widget("first_name").text, :last_name => find_widget("last_name").text, :dob => find_widget("dob").text, :telephone => find_widget("telephone").text, :email=> find_widget("email").text)
+    new_contact = add_contact_fields
+    @contacts_repository.add(new_contact)
   end
 
   private
     def find_widget(name)
       self.children.find { |child| child.object_name == name }
     end
+
+    def add_contact_fields
+      contact = Contact.new
```

Comment:
Probably the right way to do this is with the notify signal on each widget. When the widget value changes, it should automatically set the value on the 'current' contact object. For this to work you need to have a `Contact` object created when you create the window, ready for setting, and then when you save the object you simply need to push it to the repository.  This is approaching the MVC pattern.